### PR TITLE
Add '--clear-cache' flag to the sync command

### PR DESCRIPTION
**Pull request by floitsch**
We sometimes encounter corrupted caches, and this way our users can easily recover.

### CODE REVIEW

File: pkg/tpkg/registry.go [hunk 1]
```diff
@@ -337,97 +349,105 @@ func (gr *gitRegistry) Describe() string {
...
 }
 
-func (gr *gitRegistry) Load(ctx context.Context, sync bool, cache Cache, ui UI) error {
-	if sync {
-		p := gr.path
-		if gr.path == "" {
-			p = cache.PreferredRegistryPath(gr.url)
-		}
+func (gr *gitRegistry) withFileLock(ctx context.Context, cache Cache, f func(path string) error) error {
+	p := gr.path
```
Comment (floitsch) [inline pkg/tpkg/registry.go:353]: This code was previously in the 'Load' function.
Only change is the `return f(p)` in the last line.

File: pkg/tpkg/registry.go [hunk 2]
```diff
@@ -337,97 +349,105 @@ func (gr *gitRegistry) Describe() string {
...
+	return f(p)
+}
 
-		if exists {
-			err := git.Pull(p, git.PullOptions{})
-			if err != nil {
+func (gr *gitRegistry) Load(ctx context.Context, sync bool, cache Cache, ui UI) error {
+	if sync {
+		err := gr.withFileLock(ctx, cache, func(p string) error {
+			info, err := os.Stat(p)
```
Comment (floitsch) [inline pkg/tpkg/registry.go:404]: This code has only changed indentation, since it is now passed to `withFileLock`.

File: pkg/tpkg/registry.go [hunk 3]
```diff
@@ -337,97 +349,105 @@ func (gr *gitRegistry) Describe() string {
...
 		}
+	}()
+	select {
+	case <-unlocked:
+		defer m.Unlock()
+	case <-ctx.Done():
+		return fmt.Errorf("unable to acquire sync lock %s", lockP)
+	}
 
-		// Check again whether the directory exists now.
```
Comment (kasperl) [inline pkg/tpkg/registry.go:387]: Did you just drop this comment? It would go in the anonymous function expression in Load, but maybe it is obsolete?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: pkg/tpkg/registry.go (hunk 3 above)
Comment (floitsch) [reply pkg/tpkg/registry.go:387]: I felt it didn't add much, and with the split code it wasn't completely clear where the "again" came from.
It feels natural to test for the existence once you have the lock.

**Pull request merged**